support ImageMagick 6 and 7

v4hn opened the pull request:
**This patch was not tested on hardware**

I needed this module to compile with ImageMagick 7 on a system I'm using, so I spent quite some time looking for documentation/examples on the C++ accessors.
I am not 100% convinced yet this works as expected, but it's hard to test for me right now.

getConstPixels changed from PixelPacket* to Quantum* in ImageMagick 7,
so use Image::pixelColor instead. This is not as computationally efficient,
but it generalizes between the versions.

v4hn committed:
support ImageMagick 6 and 7

getConstPixels changed from PixelPacket* to Quantum* in ImageMagick 7,
so use Image::pixelColor instead. This is not as computationally efficient,
but it generalizes between the versions.

beatrizleon commented:
Hi @v4hn. Thanks for the pr. We have tested it using kinetic and imagemagick version 6.8.9.9 and we saw the robot moving in simulation. We still don't have melodic environment to test this on. Would you be able to test this soon?

v4hn commented:
Could you give me the commands to start one of your gazebo worlds and run the test for this?
I will not be able to run this environment on the real hardware any time soon.

v4hn commented:
One year later still as relevant.

I just rebased and fixed a minor code format issue involving missing whitespace.

BTW: External contributors apparently cannot see your CI reports (I had to guess the formatting error it failed on).

Commit:
Merge branch 'melodic-devel' into pr-imagemagick7
